app/controllers/concerns/foreman/controller/discovered_extensions.rb

```diff
@@ -29,6 +29,7 @@ def find_discovery_rule host
 
   # trigger the provisioning
   def perform_auto_provision host, rule
+    host = host.becomes(::Host::Managed)
```

Comment:
> @lzap I might be mistaken but from what I saw every place that called `perform_auto_provision` did it's own `becomes` so I moved it here and removed it from the other method calls.
> Is there somewhere that uses auto provision and passes the host as is?

Yeah, I realized. This stems from refactoring we did because of some
nasty bugs. But we have that covered now. It's okay.

-- 
Later,
 Lukas #lzap Zapletal